app/models/organization_resolver.rb

```diff
@@ -48,6 +49,25 @@ def organization_tag_names
     @organization&.tags || []
   end
 
+  def entities
+    [
+      @organization,
+      @email,
+      @location,
+      @organization_name,
+      @phone_number,
+      @website
+    ]
+  end
+
+  def source_names
+    entities.reduce([]) do |memo, entity|
+      next entity if entity.nil?
+      name = entity.versions.last.actor.import.source.name
+      memo.include?(name) ? memo : memo << name
+    end
```

Comment:
It's totally a readability/preference thing so I would say your call.

Regarding the complexity it's not that different with this small `n`, but it's interesting. If I understand that correctly, the complexity of current implementation is
```
1 + 2 + ... + n = n(n - 1) / 2 = O(n^2)
```
because we loop through the `memo` inside of another loop. The implementation above can be `O(n)` I believe.

But, again, readability should be the key here.